src/calva-fmt/src/format.ts

```diff
@@ -203,66 +238,160 @@ function _calculateFormatRange(
   }
 }
 
-function _formatDepth(cursor: LispTokenCursor) {
-  const cursorClone = cursor.clone();
-  cursorClone.backwardFunction(1);
-  return FormatDepthDefaults?.[cursorClone.getFunctionName()] ?? 1;
+//----------
+
+export type ReformatChange = {
+  start: number;
+  end: number;
+  text: string;
+};
+
+/** whitespace and substance. Pre-format and re-formatted text can be expressed
+ * as a series of SpacedUnit. The substance parts can be aligned and then
+ * changes in size can be translated to TextEdits.
+ */
+type SpacedUnit = [spaces: string, stuff: string];
```

Comment:
This is pretty awesome stuff! Looks lik some of it could be extracted into a module that doesn't need vscode and thus we could unit have unit tests?